clang/lib/CConv/ArrayBoundsInferenceConsumer.cpp

```diff
@@ -133,11 +133,11 @@ static bool needArrayBounds(Expr *E, ProgramInfo &Info, ASTContext *C) {
 
 static bool needArrayBounds(Decl *D, ProgramInfo &Info, ASTContext *C,
                             bool IsNtArr) {
-  CVarSet ConsVar = Info.getVariable(D, C);
-  auto &E = Info.getConstraints().getVariables();
-  for (auto CurrCVar : ConsVar) {
-    if ((!IsNtArr && needArrayBounds(CurrCVar, E)) ||
-        (IsNtArr && needNTArrayBounds(CurrCVar, E)))
+  const auto &E = Info.getConstraints().getVariables();
+  CVarOption CVar = Info.getVariable(D, C);
+  if (CVar) {
```

Comment:
Above, you write `CVar.hasValue()` -- is there a reason to do what you have, or that?

Reply:
They're the same, Option will implicitly convert to boolean. I should probably make the use of implicit conversion consistent.